frontend/src/services/notebookService.ts

```diff
@@ -290,24 +291,35 @@ export const startNotebook = (data: StartNotebookData): Promise<Notebook> => {
   });
 };
 
-export const stopNotebook = (projectName: string, notebookName: string): Promise<Notebook> => {
+export const stopNotebook = async (
+  projectName: string,
+  notebookName: string,
+): Promise<Notebook> => {
   const dateStr = new Date().toISOString().replace(/\.\d{3}Z/i, 'Z');
-  const patch: RecursivePartial<Notebook> = {
+  const notebook = await getNotebook(projectName, notebookName);
+  const patch: RecursivePartial<Notebook> = _.merge({}, notebook, {
     metadata: { annotations: { 'kubeflow-resource-stopped': dateStr } },
```

Comment:
Um... why do we care about the rest of the Notebook? We are stopping it, it should just patch... By adding this get call, you've slowed down the stopping of the Notebook as it has to wait now on fetching the notebook. 

I think the stop scenario is simply: The Notebook is running, we want to stop it, it just needs the annotation and nothing else needs to change.